### PR TITLE
Fix sending spurious Rejects when processing remote account deletion

### PR DESCRIPTION
**Pull request by ClearlyClaire**

### CODE REVIEW

File: app/services/delete_account_service.rb [hunk 1]
```diff
@@ -62,7 +63,7 @@ def call(account, **options)
   private
 
   def reject_follows!
```
Comment (Gargron) [inline app/services/delete_account_service.rb:65]: Wait, let me think... This made sense when this was SuspendAccountService, but do we actually *ever* need to do this when we're deleting now?

Comment (ThibG) [reply app/services/delete_account_service.rb:65]: Normally, this is already done in `SuspendAccountService`, but some codepaths skip this, such as blocking a domain altogether.

Comment (Gargron) [reply app/services/delete_account_service.rb:65]: OK, makes sense.

**Comment by Gargron** [comment]
We've got `:skip_side_effects` and now also `:skip_activitypub`, this will likely cause confusion in the future. Can we:

- Make the behaviour part of already existing `:skip_side_effects` or
- Make the name more specific like `:skip_reject_follows`?

**Comment by ThibG** [comment]
`:skip_side_effects` also skips sending streaming events, which we don't want to do here
`:skip_reject_follows` might not be the best because I also want to undo outgoing follows here

**Comment by ThibG** [comment]
`:skip_side_effects` could imply `:skip_activitypub` but those two should be different things

**Comment by gigamaster** [comment]
mastodon-social-blocked

#mastodonsocial 

Account suspended because of "Conspiracy theories"
How dumb is this... I cannot even post a link to my own account ?

There's no option to submit an "unblock appeal",
and the email sent says you keep some personal data.

You should consider there are non-EU countries that have data protection laws considered adequate by the EU, example Switzerland.

**Review by Gargron — APPROVED** [review]

**Pull request merged**